src/proc_diskstats.c

```diff
@@ -57,7 +67,7 @@ struct disk *get_disk(unsigned long major, unsigned long minor) {
 
 	d->major = major;
 	d->minor = minor;
-	d->partition_id = -1;
+	d->type = DISK_TYPE_UNKNOWN;
```

Comment:
You can start with DISK_TYPE_PHYSICAL and remove the `if(d->type == DISK_TYPE_UNKNOWN)` below and the `#define DISK_TYPE_UNKNOWN`